Show previews and sort mode for tags

tijmenb opened the pull request:
This PR:

- Adds previews for lists on the show pages of `MainstreamBrowsePage` and `Topic`.
- Adds a little icon to all relevant pages to show wether the tag is alphabetical or curated.

Hopefully this will it easier to grasp the different "sort modes" for a tag.

Needs some wordsmithery.

## Before

![screen shot 2015-06-10 at 13 20 58](https://cloud.githubusercontent.com/assets/233676/8082241/aa853d56-0f73-11e5-8e1a-bf0822b91604.png)
![screen shot 2015-06-10 at 13 21 00](https://cloud.githubusercontent.com/assets/233676/8082237/aa8366b6-0f73-11e5-9594-a3b1a5dc3fd1.png)
![screen shot 2015-06-10 at 13 21 02](https://cloud.githubusercontent.com/assets/233676/8082239/aa83d0ce-0f73-11e5-86c8-365f9528a6c2.png)

## After

![screen shot 2015-06-10 at 13 21 13](https://cloud.githubusercontent.com/assets/233676/8082238/aa837da4-0f73-11e5-994d-15b73bfdcd80.png)
![screen shot 2015-06-10 at 13 21 18](https://cloud.githubusercontent.com/assets/233676/8082236/aa831d28-0f73-11e5-94eb-80b8e0506fc0.png)

![screen shot 2015-06-10 at 13 21 23](https://cloud.githubusercontent.com/assets/233676/8082240/aa83f9b4-0f73-11e5-9575-7c0068717a0a.png)
(curated)

![screen shot 2015-06-10 at 13 22 58](https://cloud.githubusercontent.com/assets/233676/8082260/d9c5786a-0f73-11e5-8110-a0889f94cc02.png)
(alphabetized)

alextea commented:
I wonder if rather than using icons we could just add "(curated)" to the title of the curated lists?

I think this would be better as there would be less visual clutter, and it would be more obvious than the icons alone.

I'd like to hear what the content team think though as they will be using this.

tijmenb commented:
Updated PR after some feedback. Removed the second icon from the show page and added a tooltip that tells users whether the tag is curated or a-to-z.

Updated screenshot:

![screen shot 2015-06-11 at 13 13 42](https://cloud.githubusercontent.com/assets/233676/8106711/b365ecea-103b-11e5-984d-6fb35e3ba042.png)

rboulton commented:
Code looks good. :sparkles:
